M2/Macaulay2/packages/ResolutionsOfStanleyReisnerRings.m2

```diff
@@ -0,0 +1,592 @@
+    SeeAlso
+        barycentricSubdivision
+        SimplicialComplexes
+        Posets 
+        SimplicialDecomposability
+///
+
+doc ///
+    Key
+        uSOP
+    Headline
+        Universal System of Parameters (USOP)
+    Usage 
+        uSOP D
+        uSOP F
+    Inputs
+        D:SimplicialComplex
+        F:List
+            a list of facets defining a simplicial complex
+    Outputs
+        Theta:List
+         a list of elements from the Stanley-Reisner ring of {\tt D}
+    Description
+        Text
+            The following example uses the simplicial complex consisting of a triangle of dimension 2 attached to an edge and a isolated vertex.
+        Example
+            S = ZZ/2[a..e];
+            F = {a*b*c,c*d,e}
+            D = simplicialComplex F
+            uSOP D
+            uSOP F
+    SeeAlso
+        SimplicialComplexes
+/// 
+
+doc ///
+      Key
+        sumBetti
+    Headline
+        Computes the sum of Betti numbers.
+    Usage
+        sumBetti M
+    Inputs
+        M:Module
+    Outputs
+         :ZZ
+             sum of the Betti numbers of the resolution of {\tt M} 
+    Description 
+        Example
+            S = QQ[a..f];
+            m = matrix{{a,b,d,e},{b,c,e,f}} 
+            M = coker m
+            sumBetti M
+    SeeAlso
+        Module 
+///
+
+doc ///
+      Key
+        equalBettiTally
+    Headline
+        Computes the Betti table of two modules and compares for equality 
+    Usage
+        equalBettiTally(M,N)
+    Inputs
+        M:Module
+        N:Module
+    Outputs
+        :Boolean
+    Description 
+        Example
+            S = QQ[a..f];
+            m = matrix{{a,b,d,e},{b,c,e,f}} 
+            M = coker m
+            n = matrix{{a,b},{c,d},{e,f}}
+            N = coker n
+            equalBettiTally(M,N)
+    SeeAlso
+        Module 
+///
+
+doc ///
+      Key
+        universalPresentation 
+    Headline
+        A presentation of the Stanley-Reisner ring over its universal parameter ring
+    Usage
+        universalPresentation D 
+        universalPresentation F 
+    Inputs
+        D:SimplicialComplex
+        F:List
+            A list of facets defining a simplicial complex {\tt D}
+    Outputs
+        M:Module 
+            A graded presentation of the Stanley-Reisner ring over its universal parameter ring
+    Description 
+        Text
+            Note that {\tt M} is N-graded.
+        Example
+		        S = QQ[a..e];
+            F = {a*b*c,c*d,e}
+            D = simplicialComplex F
+            universalPresentation D 
+            M = universalPresentation F 
+            degrees M
+    SeeAlso
+        Module 
+        uSOP
+///
+
+doc ///
+      Key
+        colorfulPresentation 
```

Comment:
add keys for (colorfulPresentation,List) and (colorfulPresentation,SimplicialComplex), too